pkg/apis/core/types.go

```diff
@@ -2831,6 +2877,20 @@ type ContainerStatus struct {
 	// +featureGate=ResourceHealthStatus
 	// +optional
 	AllocatedResourcesStatus []ResourceStatus
+	// Lifecycle is used to show the container's effective stop signal in the Status
+	// +featureGate=ContainerStopSignals
+	// +optional
+	Lifecycle ContainerLifecycleStatus
+}
+
+// ContainerLifecycleStatus represents the effective stop signal used by the container runtime.
```

Comment:
Picking at words:

```
// ContainerLifecycleStatus reports information about the lifecycle management of one container.
type ContainerLifecycleStatus struct {
    // StopSignal is the signal which will be sent to this container when it is stopped.  This might be
    // the stop signal specified by the user, the signal specified in the container image, or the default
    // stop signal of the container runtime on this node.
```
```

Reply:
Done in https://github.com/kubernetes/kubernetes/pull/130556/commits/6041eba787c31598f519ab4de14edd2883500431.